### PR TITLE
Normalize HunYuanVL's legacy field aliases via attribute_map

### PR DESCRIPTION
**Pull request by hmellor**
[CPU CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48261) [GPU run-slow](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48261)

# What does this PR do?

`HunYuanVLTextConfig`'s `__post_init__` carries this comment:

> Legacy aliases (`pad_id`, `attention_head_dim`, `org_vocab_size`) are normalized onto canonical fields by the base `__setattr__` via `attribute_map`, so no manual translation is needed here.

but `attribute_map` only ever listed `pad_id`. So `attention_head_dim` and `org_vocab_size` fell through `PreTrainedConfig.__post_init__`'s catch-all `setattr` loop and were re-attached as untyped attributes, duplicating `head_dim` / `vocab_size` and round-tripping back out through `save_pretrained`.

This PR adds the two missing entries so the code matches what the docstring already claims, and extends the "flat checkpoint" fold in `HunYuanVLConfig.__post_init__` to also recognize alias names, so a top-level `attention_head_dim` lands in `text_config` rather than sticking to the root config.

Every public Tencent checkpoint that persists these aliases (`tencent/HunyuanOCR`, `tencent/Hunyuan-7B-Instruct`, `tencent/Hunyuan-A13B-Instruct`) stores them with the same value as the canonical field, so nothing changes semantically — the alias stays readable via `attribute_map`, it just isn't stored (or serialized) twice.

A test covers the round-trip.

<details>
<summary>Originally this PR also dropped the legacy MoE/MLA keys</summary>

The `tencent/HunyuanOCR` config used to persist the routing and multi-head-latent-attention hyperparameters it inherited from the Tencent MoE codebase — most visibly `num_experts=1`, which made a dense-only OCR model advertise itself as a 1-expert MoE to anything downstream that sniffs the config. The first commit stripped them on load.

Tencent has since [updated the checkpoint](https://huggingface.co/tencent/HunyuanOCR/blob/main/config.json) to remove those fields, so that workaround is no longer needed and has been reverted (thanks @zucchini-nlp for chasing it upstream).

</details>

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request)?
- [x] Did you write any new necessary tests?

## Who can review?

@zucchini-nlp

### CODE REVIEW

**Comment by zucchini-nlp** [comment]
> HunYuanVL is a dense-only variant and declares none of these as config fields, so PreTrainedConfig.__post_init__ falls through to its "additional attributes without default values" branch and setattrs every one of them onto the config. 

A bit confused, so the model is non-MoE and the config fields for `experts` are not used anywhere (in transformers or in remote code)? If I got it right, I think we'd better ask Tencent to drop unused config fields from json file. They do reply on slack so we have a chance to get it fixed upstream on the Hub

Taking a look at second one and wll comment in code

File: src/transformers/models/hunyuan_vl/modular_hunyuan_vl.py [hunk 1]
```diff
@@ -303,18 +340,25 @@ def __post_init__(self, **kwargs):
         # nested `text_config` block) we fold the recognized text-side keys into the text config payload. This keeps
         # ``HunYuanVLConfig.from_pretrained(...)`` working with both the upstream nested layout and the existing
         # public OCR checkpoints.
-        text_keys = set(self.sub_configs["text_config"].__dataclass_fields__) | {"rope_scaling", "rope_theta"}
+        text_config_class = self.sub_configs["text_config"]
+        text_keys = (
+            set(text_config_class.__dataclass_fields__)
+            | set(text_config_class.attribute_map)
+            | {"rope_scaling", "rope_theta"}
+        )
```
Comment (zucchini-nlp) [inline src/transformers/models/hunyuan_vl/modular_hunyuan_vl.py:314]: oke, this makes sense to me. Ugly but what can we do with released files 😢

**Review by zucchini-nlp — COMMENTED** [review]
The second point is fine by me, and lmk if the first one is rather a `config.json` problem than "transformers code". I can ping the authors in slack and ask to delete unused fields

**Comment by hmellor** [comment]
> A bit confused, so the model is non-MoE and the config fields for experts are not used anywhere (in transformers or in remote code)?

Yeah that's right. The problem is that their existence makes introspecting the config to determine what the model is a difficult task. It's impossible to know if the expert specific configs are added by accident or not.

**Comment by zucchini-nlp** [comment]
Asked them in slack and tagged you

**Comment by zucchini-nlp** [comment]
they removed it now, yaya! Lets revert #1 fixes and merge

**Comment by hmellor** [comment]
Reverted the MoE/MLA key dropping in eba153e — the updated `tencent/HunyuanOCR` config is clean, thanks for chasing it upstream 🙏

What's left is just the `attribute_map` fix: `attention_head_dim` and `org_vocab_size` were missing from it, despite the comment right above `__post_init__` claiming they were mapped, so they were still landing as duplicate untyped attributes alongside `head_dim` / `vocab_size`.

**Review by zucchini-nlp — APPROVED** [review]
Yep, it makes sense to me! IIRC they couldn't or didn't want to rename config keys, sadly
